samples/boards/nrf/nrf_led_matrix/src/main.c

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2021, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <drivers/display.h>
+
+void main(void)
+{
+	printk("nRF LED matrix sample on %s\n", CONFIG_BOARD);
+
+	int ret;
+	uint8_t buf[5];
+	const struct display_buffer_descriptor buf_desc = {
+		.buf_size = sizeof(buf),
+		.width    = 5,
+		.height   = 5,
+		.pitch    = 8,
+	};
+	const struct device *dev = DEVICE_DT_GET(DT_INST(0, nrf_led_matrix));
+
+	if (!dev) {
+		printk("No display device\n");
+		return;
+	}
+
+	for (;;) {
+		printk("Started\n");
+
+		ret = display_set_brightness(dev, 0x7F);
+		if (ret < 0) {
+			printk("display_set_brightness failed: %u/%d\n",
+				__LINE__, ret);
+		}
+
+		buf[0] = 0xA8;
+		buf[1] = 0x40;
+		buf[2] = 0xA8;
+		buf[3] = 0x10;
+		buf[4] = 0xA8;
+		ret = display_write(dev, 0, 0, &buf_desc, buf);
+		if (ret < 0) {
+			printk("display_write failed: %u/%d\n",
+				__LINE__, ret);
+		}
+
+		ret = display_blanking_off(dev);
+		if (ret < 0) {
+			printk("display_blanking_off failed: %u/%d\n",
+				__LINE__, ret);
+		}
```

Comment:
I suggest to move show blocks to own function led_matrix_show_foo(), led_matrix_show_bar()...